bundler/helpers/lib/functions/parent_dependency_resolver.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+module Functions
+  class ParentDependencyResolver
+    def initialize(dependency_name:, target_version:, lockfile_name:)
+      @dependency_name = dependency_name
+      @target_version = target_version
```

Comment:
```suggestion
      @target_version = Gem::Version.new(target_version)
```

Idea: the only usage is as a `Gem::Version`, just store as that.